fig/service.py

```diff
@@ -629,6 +634,25 @@ def split_port(port):
     return internal_port, (external_ip, external_port or None)
 
 
+def build_extra_hosts(extra_hosts_config):
+    if extra_hosts_config is None:
+        return None
+
+    if isinstance(extra_hosts_config, basestring):
+        extra_hosts_config = [extra_hosts_config]
+
+    extra_hosts_dict = {}
+    for extra_hosts_line in extra_hosts_config:
+        if isinstance(extra_hosts_line, dict):
+            # already interpreted as a dict (depends on pyyaml version)
+            extra_hosts_dict.update(extra_hosts_line)
+        else:
+            # not already interpreted as a dict
+            extra_hosts_dict.update(extra_hosts_line.split(':'))
```

Comment:
Actually this introduces a bug

I just commited a fix + some unit tests, could you review them please ?
https://github.com/tdesvenain/fig/commit/3e6c7cd5a49a52f68184a3d784066d00ab377c1d

2015-02-25 13:27 GMT+01:00 ddunwoody <notifications@github.com>:

> In fig/service.py
> <https://github.com/docker/fig/pull/848#discussion_r25335828>:
>
> > @@ -629,6 +634,25 @@ def split_port(port):
> >      return internal_port, (external_ip, external_port or None)
> >
> >
> > +def build_extra_hosts(extra_hosts_config):
> > +    if extra_hosts_config is None:
> > +        return None
> > +
> > +    if isinstance(extra_hosts_config, basestring):
> > +        extra_hosts_config = [extra_hosts_config]
> > +
> > +    extra_hosts_dict = {}
> > +    for extra_hosts_line in extra_hosts_config:
> > +        if isinstance(extra_hosts_line, dict):
> > +            # already interpreted as a dict (depends on pyyaml version)
> > +            extra_hosts_dict.update(extra_hosts_line)
> > +        else:
> > +            # not already interpreted as a dict
> > +            extra_hosts_dict.update(extra_hosts_line.split(':'))
>
> I believe line 651 should read:
>
> extra_hosts_dict.update([extra_hosts_line.split(':')])
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/fig/pull/848/files#r25335828>.
>



-- 
Thomas Desvenain

Téléphone : 09 51 37 35 18